puppetserver/enc/foreman.rb

```diff
@@ -0,0 +1,470 @@
+#!/usr/bin/env ruby
```

Comment:
I see a few problems with this:
* the source is GPL-3.0+, we are Apache-2. I don't know if we're allowed to vendor this
* the two files have no indication that they are coming from the foreman project
* How do we keep track of upstream changes? Would it make more sense to pull the files in when we built the container?

Reply:
I see

- I have no idea :)
- Would it help with a disclaimer at the top?
- Those two files haven't changed in years, so in practice it probably won't be an issue. But I understand the concern.

Reply:
maybe adding a script, that if $SOMETHING is set, it curls in the upstream files from their sources?

Reply:
maybe adding a script, that if $SOMETHING is set, it curls in the upstream files from their sources?

Reply:
I run this in an air gapped environment, so that wouldn't work for me, unfortunately

Reply:
can you please provide the upstrem urls from where the files come from? so that we know how/where to update, look for updates?

Reply:
I picked it out from The Foreman's own git repo for their contaiers [here](https://github.com/theforeman/puppet-puppetserver_foreman/blob/master/files/enc.rb).

I can't seem to find other official sources. It is, however, the one that's installed when you install Foreman and Puppet using the installer.